diff --git a/scripts/linux/package-linux.sh b/scripts/linux/package-linux.sh
index 6d90092..036ebc7 100755
--- a/scripts/linux/package-linux.sh
+++ b/scripts/linux/package-linux.sh
@@ -3,7 +3,7 @@ echo "cur bin $cur__bin"
 rm -rf _release
 rm -rf _staging
 
-mkdir -p _release/linux
+mkdir -p _release/linux _release/Onivim2.AppDir/{usr/bin,usr/share,usr/lib}
 mkdir -p _staging
 
 cp -r $cur__bin _release/linux
@@ -29,9 +29,12 @@ cp ThirdPartyLicenses.txt _release/Onivim2.AppDir/ThirdPartyLicenses.txt
 
 cp _release/linux/bin/*.* _release/Onivim2.AppDir/usr/bin
 cp _release/linux/bin/Oni2 _release/Onivim2.AppDir/usr/bin/Oni2
+cp _release/linux/bin/Oni2_editor _release/Onivim2.AppDir/usr/bin/Oni2_editor
 
 cp vendor/ripgrep-v0.10.0/linux/rg _release/Onivim2.AppDir/usr/bin/rg
 cp vendor/node-v14.15.4/linux-x64/node _release/Onivim2.AppDir/usr/bin/node
+cp vendor/reason-language-server/bin.native.linux _release/Onivim2.AppDir/usr/bin/rls
+cp $JPEG_LIB_PATH/* _release/Onivim2.AppDir/usr/lib
 
 ls _release/Onivim2.AppDir/usr/share
 
@@ -41,5 +44,5 @@ cp -r node/ _release/Onivim2.AppDir/usr/share
 
 rm -f _release/Onivim2.AppDir/usr/bin/setup.json
 
-ARCH=x86_64 _staging/appimagetool-x86_64.AppImage _release/Onivim2.AppDir _release/Onivim2-x86_64.AppImage
+ARCH=x86_64 _staging/appimagetool-x86_64.AppImage --appimage-extract-and-run _release/Onivim2.AppDir _release/Onivim2-x86_64.AppImage
 
diff --git a/scripts/release.js b/scripts/release.js
index 100be84..d0d4c97 100644
--- a/scripts/release.js
+++ b/scripts/release.js
@@ -106,7 +106,7 @@ const updateIcon = (rcedit, exe, iconFile) => {
 }
 
 if (process.platform == "linux") {
-    const result = cp.spawnSync("esy", ["bash", "-c", "scripts/linux/package-linux.sh"], {
+    const result = cp.spawnSync("esy", ["@release", "bash", "-x", "-c", "scripts/linux/package-linux.sh"], {
         cwd: process.cwd(),
         env: process.env,
         stdio: "inherit",
